engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala

```diff
@@ -314,15 +315,27 @@ final case class WorkflowExecutionActor(workflowId: WorkflowId,
     case Event(FailedNonRetryableResponse(jobKey, reason, returnCode), stateData) =>
       pushFailedJobMetadata(jobKey, returnCode, reason, retryableFailure = false)
       saveUnsuccessfulJobResults(jobKey, returnCode, reason)
-      context.parent ! WorkflowExecutionFailedResponse(stateData.executionStore, stateData.outputStore, List(reason))
       val mergedStateData = stateData.mergeExecutionDiff(WorkflowExecutionDiff(Map(jobKey -> ExecutionStatus.Failed)))
-      goto(WorkflowExecutionFailedState) using mergedStateData.removeBackendJobExecutionActor(jobKey)
+                                     .removeBackendJobExecutionActor(jobKey)
+      if (workflowDescriptor.getWorkflowOption(WorkflowFailureMode).contains(ContinueWhilePossible.toString)) {
+        mergedStateData.workflowCompletionStatus match {
+          case Some(completionStatus) if completionStatus == Failed =>
+            context.parent ! WorkflowExecutionFailedResponse(stateData.executionStore, stateData.outputStore, List(reason))
+            goto(WorkflowExecutionFailedState) using mergedStateData
+          case _ =>
+            stay() using startRunnableScopes(mergedStateData)
+        }
+      } else {
+        context.parent ! WorkflowExecutionFailedResponse(stateData.executionStore, stateData.outputStore, List(reason))
+        goto(WorkflowExecutionFailedState) using mergedStateData
+      }
     case Event(FailedRetryableResponse(jobKey, reason, returnCode), stateData) =>
       workflowLogger.warn(s"Job ${jobKey.tag} failed with a retryable failure: ${reason.getMessage}")
       pushFailedJobMetadata(jobKey, None, reason, retryableFailure = true)
       handleRetryableFailure(jobKey, reason, returnCode)
     case Event(JobInitializationFailed(jobKey, reason), stateData) =>
       pushFailedJobMetadata(jobKey, None, reason, retryableFailure = false)
+      context.parent ! WorkflowExecutionFailedResponse(stateData.executionStore, stateData.outputStore, List(reason))
```

Comment:
I guess I was interpreting job initialization as a hard-failure but I think you're right that it should not be.